analyzers/src/SonarAnalyzer.Common/Rules/Hotspots/DisablingRequestValidationBase.cs

```diff
@@ -69,5 +80,79 @@ protected DisablingRequestValidationBase(System.Resources.ResourceManager rspecR
                 },
                 SymbolKind.NamedType,
                 SymbolKind.Method);
+
+            context.RegisterCompilationAction(c =>
+            {
+                analyzerConfiguration.Initialize(c.Options);
+                if (!analyzerConfiguration.IsEnabled(DiagnosticId))
+                {
+                    return;
+                }
+                foreach (var fullPath in Directory.GetFiles(rootPath, "web.config", SearchOption.AllDirectories).Select(p => Path.GetFullPath(p)))
+                {
+                    var webConfig = File.ReadAllText(fullPath);
+                    if (webConfig.Contains("<system.web>") && ParseXDocument(webConfig) is { } doc)
+                    {
+                        ReportValidateRequest(doc, fullPath, c);
+                        ReportRequestValidationMode(doc, fullPath, c);
+                    }
+                }
+            });
+        }
+
+        private void ReportValidateRequest(XDocument doc, string webConfigPath, CompilationAnalysisContext c)
+        {
+            foreach (var pages in doc.XPathSelectElements("configuration/system.web/pages"))
+            {
+                if (pages.Attribute("validateRequest") is { } validateRequest
+                    && validateRequest.Value == "false"
```

Comment:
Not sure about it but `False` can also be correctly parsed, right? Shouldn't we do a case insensitive check or actually parse the value?

I ask about this since, as far as I can tell, the framework uses behind [NameValueCollection](https://docs.microsoft.com/en-us/dotnet/api/system.collections.specialized.namevaluecollection?view=netframework-4.8) when parsing the file, which uses a case insensitive comparer.

Reply:
good point (even if XML complains only "false" respects the schema boolean type)